en/computer_vision/collision_prevention.md

```diff
@@ -30,16 +27,38 @@ The user is notified through *QGroundControl* while *Collision Prevention* is ac
 
 ## PX4 (Software) Setup
 
-Set the following [parameter](../advanced_config/parameters.md) in *QGroundControl*:
+Enabled collision prevention by [setting the following parameter](../advanced_config/parameters.md) in *QGroundControl*:
 
-* [MPC_COL_PREV_D](../advanced_config/parameter_reference.md#MPC_COL_PREV_D) - Set the minimum allowed distance (the closest distance that the vehicle can approach the obstacle). Set negative to disable *collision prevention*.
+* [MPC_COL_PREV_D](../advanced_config/parameter_reference.md#MPC_COL_PREV_D) - Set the minimum allowed distance (the closest distance that the vehicle can approach the obstacle).
+  Set negative to disable *collision prevention*.
 
   This should be tuned for both the *desired* minimal distance and likely speed of the vehicle.
 
+If you are using a distance sensor attached to your flight controller for collision prevention, it will need to be [attached and configured](#rangefinder) as described in the next section.
+If you are using a companion computer to provide obstacle information see [companion setup](#companion).
+
+
+## PX4 Distance Sensor {#rangefinder}
+
+At time of writing PX4 allows you to use the [Lanbao PSK-CM8JL65-CC5](../sensor/cm8jl65_ir_distance_sensor.md) IR distance sensor for collision prevention "out of the box", with minimal additional configuration:
+- First [attach and configure the sensor](../sensor/cm8jl65_ir_distance_sensor.md), and enable collision prevention (as described above, using `MPC_COL_PREV_D`).
+- Set the sensor orientation using [SENS_CM8JL65_R_0](../advanced_config/parameter_reference.md#SENS_CM8JL65_R_0).
+
+<!-- ROTATION_FORWARD_FACING - Does it matter what angles? - ie is collision prevention active in 3 D? -->
 
-## Companion Setup
+Other sensors may be enabled, but this requires modification of driver code to set the sensor orientation and field of view.
+- Attach and configure the distance sensor on a particular port (see [sensor-specific docs](../sensor/rangefinders.md)) and enable collision prevention using `MPC_COL_PREV_D`.
+- Modify the driver to set the orientation. 
+  This should be done by mimicking the `SENS_CM8JL65_R_0` parameter (though you might also hard-code the orientation in the sensor *module.yaml* file to something like `sf0x start -d ${SERIAL_DEV} -R 25` - where 25 is equivalent to `ROTATION_DOWNWARD_FACING`).
```

Comment:
@mrivi This is worded as it is, because we'd really like to be able to support all sensors "out of the box", so this encourages a good copy. 
If instructions are clear enough we might be able to encourage @mcsauder to backfill this.

Reply:
Yes, I agree. I didn't want to add two many useless parameters. However this PR https://github.com/PX4/Firmware/pull/12567/files seems to go already in the direction we need. 
I really would like to have the rotation paramter to be common for all the distance sensor and not specific to one.

Reply:
There is also this rangefinder, which is specifically for collision prevention, and includes per-sensor rotation parameters already: https://github.com/PX4/Firmware/pull/12147

I'm going to be testing them this week.

Reply:
@mrivi I guess the advantage of individual params is that the driver doesn't have to know anything else about the system config - e.g. if there are multiple sensors.

@mhkabir Mappydot looks very cool but appears to lack [module](http://dev.px4.io/master/en/middleware/modules_driver_distance_sensor.html) or [end user](https://docs.px4.io/v1.9.0/en/sensor/rangefinders.html) docs? Do you think you could help (or help me) create that when you have this tested?

Also to use a slightly different naming convention for the rotation param SENS_MPDTn_ROT vs SENS_CM8JL65_R_n (where `n`, is sensor number). Nice to be consistent (personally I like the use of ROT.